src/fluree/db/datatype.cljc

```diff
@@ -46,37 +46,35 @@
    "http://www.w3.org/2001/XMLSchema#hexBinary"          const/$xsd:hexBinary
    "http://www.w3.org/2001/XMLSchema#base64Binary"       const/$xsd:base64Binary})
 
+(def iso8601-offset-pattern
+  "(Z|(?:[+-][0-9]{2}:[0-9]{2}))?")
+
 (def iso8601-date-component-pattern
   "This is slightly more forgiving than the xsd:date spec:
   http://books.xmlschemata.org/relaxng/ch19-77041.html
   Note there is no need to be extra strict with the numeric ranges in here as
   the java.time constructors will take care of that for us."
-  "((?:-)?[0-9]{4})-([0-9]{1,2})-([0-9]{1,2})")
+  "((?:-)?[0-9]{4})-([0-9]{2})-([0-9]{2})")
 
 (def iso8601-date-pattern
   "Defines the pattern for dates w/o times where an offset is still allowed on
   the end."
-  (str iso8601-date-component-pattern "(Z|(?:[+-][0-9]{1,2}:[0-9]{2}))?"))
+  (str iso8601-date-component-pattern iso8601-offset-pattern))
 
 (def iso8601-date-re
   (re-pattern iso8601-date-pattern))
 
 (def iso8601-time-pattern
-  "This is slightly more forgiving than the xsd:time spec:
-  http://books.xmlschemata.org/relaxng/ch19-77311.html
-  Note there is no need to be extra strict with the numeric ranges in here as
-  the java.time constructors will take care of that for us."
-  "([0-9]{1,2}):([0-9]{1,2}):([0-9]{1,2})(Z|(?:[+-][0-9]{1,2}:[0-9]{2}))?")
+  (str #?(:clj  "([0-9]{2}):([0-9]{2}):([0-9]{2})(?:\\.([0-9]{1,9}))?"
```

Comment:
This is the heart of the fix, adding an optional capture group to capture subsecond precision: `(?:\\.([0-9]{1,9}))?`